hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/utils/db/RDBMetrics.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.hadoop.hdds.utils.db;
+
+import org.apache.hadoop.metrics2.MetricsSystem;
+import org.apache.hadoop.metrics2.annotation.Metric;
+import org.apache.hadoop.metrics2.lib.DefaultMetricsSystem;
+import org.apache.hadoop.metrics2.lib.MutableCounterLong;
+
+import com.google.common.annotations.VisibleForTesting;
+
+/**
+ * Class to hold RocksDB metrics.
+ */
+public class RDBMetrics {
+
+  private static final String SOURCE_NAME =
+      RDBMetrics.class.getSimpleName();
+
+  public RDBMetrics() {
+  }
+
+  public static RDBMetrics create() {
+    MetricsSystem ms = DefaultMetricsSystem.instance();
+    return ms.register(SOURCE_NAME,
+        "Rocks DB Metrics",
+        new RDBMetrics());
+  }
+
+  private @Metric MutableCounterLong numDBKeyMayExistChecks;
+  private @Metric MutableCounterLong numDBKeyMayExistMisses;
+
+
+  public void incNumDBKeyMayExistChecks() {
+    numDBKeyMayExistChecks.incr();
+  }
+
+  public void incNumDBKeyMayExistMisses() {
+    numDBKeyMayExistMisses.incr();
+  }
+
+
+  @VisibleForTesting
+  public long getNumDBKeyMayExistChecks() {
+    return numDBKeyMayExistChecks.value();
+  }
+
+  @VisibleForTesting
+  public long getNumDBKeyMayExistMisses() {
+    return numDBKeyMayExistMisses.value();
+  }
+
+  public void unRegister() {
+    MetricsSystem ms = DefaultMetricsSystem.instance();
```

Comment:
This should be called during OM/SCM/Recon stop. As this common class for all of them.

Reply:
Good catch. Will add it.

Reply:
This is used in OM and Recon. Will add the unregister step in them.

Reply:
SCMMetadataStoreRDBImpl also uses RDBStore. SCMMetadataStoreRDBImpl is used in StorageContainerManager.

Reply:
Thanks @bharatviswa504. That should be covered with this change since StorageContainerManager#stop --> SCMMetadataStore#stop --> DBStore.close() -> RDBMetrics.unregister(). Please correct me if I am wrong.

Reply:
Ya right, I got confused with the comment it is used in OM and Recon.